src/routes/Root/Root.tsx

```diff
@@ -104,10 +104,11 @@ const Root = () => {
   };
 
   const changeLanguage = async (lang: LangCode) => {
-    setLangMenuOpen(false);
+    const newLocation = resolveLocalizedUrl(lang);
     await i18n.changeLanguage(lang);
-    setLanguage(lang);
-    navigate(`/${i18n.language}`);
+    setLangMenuOpen(false);
+    // refresh to load app locales
+    location.assign(`${newLocation}`);
```

Comment:
Is this necessary? Won't this reload the entire application again?

Reply:
Me and Markus tried bunch of things to make it work without assign, but we were unable to find a working solution. The issue seems to be that the component tree doesn't completely rerender when language is changed, and it causes `NavLink`'s to have the wrong href. It's tied to how `NavLink` resolves the `to` prop relatively. I tried to create a `LanguageProvider` that stores the language and handles the language change, and wrapped the `App` component into it, which should cause the whole app to rerender, but that didn't help.

There was one "working" and dumb solution: Add delay and call stuff twice in the changeLanguage function 🙃:
```
await i18n.changeLanguage(lang);
navigate(newLocation);
await new Promise((res) => setTimeout(res, 500));
await i18n.changeLanguage(lang);
navigate(newLocation);
```

So we decided to leave the location.assign for now.